File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/wrapper/utils/event/click/ClickEvent.java
package rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.utils.event.click;


import rbq.wtf.lycoris.client.wrapper.MapEnum;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapField;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapperClass;
import rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.IWrapper;

import java.lang.reflect.Field;

@WrapperClass(mcpName = "net.minecraft.event.ClickEvent",targetMap = MapEnum.Srg1_8_9)
public class ClickEvent extends IWrapper {
    @WrapField(mcpName = "action",targetMap = MapEnum.Srg1_8_9)
    public static Field action;
    @WrapField(mcpName = "value",targetMap = MapEnum.Srg1_8_9)
    public static Field value;
    public ClickEvent(Object obj) {
        super(obj);
    }
    public Enum getAction(){
        return (Enum) getField(action);
    }
    public String getValue(){
        return (String) getField(value);
    }
}

File: src/main/kotlin/rbq/lycoris/client/wrapper/wrappers/util/event/click/ClickEvent.kt
package rbq.lycoris.client.wrapper.wrappers.util.event.click

import rbq.lycoris.client.wrapper.IWrapper
import rbq.lycoris.client.wrapper.MapEnum
import rbq.lycoris.client.wrapper.annotation.WrapField
import rbq.lycoris.client.wrapper.annotation.WrapperClass
import java.lang.reflect.Field

@WrapperClass(mcpName = "net.minecraft.event.ClickEvent", targetMap = MapEnum.VANILLA189)
class ClickEvent(obj: Any) : IWrapper(obj) {
    companion object {
        @WrapField(mcpName = "action", targetMap = MapEnum.VANILLA189)
        lateinit var action: Field

        @WrapField(mcpName = "value", targetMap = MapEnum.VANILLA189)
        lateinit var value: Field
    }

    val action: Enum<*>
        get() = getField(Companion.action) as Enum<*>
    val value: String
        get() = getField(Companion.value) as String
}